src/representative.jl

```diff
@@ -9,17 +18,106 @@ used for each representative period. In addition, each representative period
 is attributed to it.
 
+!!! note
+    - All representative periods must use the same type for the `TimeStructure`.
+    - If the field `period_share` is not specified, it assigns the same probability to each
+      representative period.
+    - It is possible that `sum(period_share)` is larger or smaller than 1. This can lead to
+      problems in your application. Hence, it is advised to scale it. We provide currently a
+      warning as it would correspond to a breaking change.
+    - If you include [`OperationalScenarios`](@ref) in your time structure, it is important
+      that you the scenarios are within the representative periods, and note the other way.
+
 ### Example
 ```julia
 # A year represented by two days with hourly resolution and relative shares of 0.7 and 0.3
-periods = RepresentativePeriods(2, 8760, [0.7, 0.3], [SimpleTimes(24, 1), SimpleTimes(24,1)])
+RepresentativePeriods(8760, [0.7, 0.3], [SimpleTimes(24, 1), SimpleTimes(24,1)])
+RepresentativePeriods(8760, [0.7, 0.3], SimpleTimes(24, 1))
+
+# A year represented by two days with hourly resolution and relative shares of 0.5
+RepresentativePeriods(2, 8760, SimpleTimes(24, 1))
+RepresentativePeriods(8760, [SimpleTimes(24, 1), SimpleTimes(24,1)])
 ```
 """
-struct RepresentativePeriods{S,T,OP<:TimeStructure{T}} <: TimeStructure{T}
-    len::Int
+struct RepresentativePeriods{S<:Duration,T,OP<:TimeStructure{T}} <:
+       TimeStructure{T}
+    len::Integer
     duration::S
     period_share::Vector{Float64}
     rep_periods::Vector{OP}
+    function RepresentativePeriods(
+        len::Integer,
+        duration::S,
+        period_share::Vector{Float64},
+        rep_periods::Vector{OP},
+    ) where {S<:Duration,T,OP<:TimeStructure{T}}
+        if len > length(period_share)
+            throw(
+                ArgumentError(
+                    "The length of `period_share` cannot be less than the field `len` of `RepresentativePeriods`.",
+                ),
+            )
+        elseif len > length(rep_periods)
+            throw(
+                ArgumentError(
+                    "The length of `rep_periods` cannot be less than the field `len` of `RepresentativePeriods`.",
+                ),
+            )
+        elseif sum(period_share) > 1 || sum(period_share) < 1
+            @warn(
+                "The sum of the `period_share` vector is given by $(sum(period_share)). " *
+                "This can lead to unexpected behaviour."
+            )
+        end
+        return new{S,T,OP}(len, duration, period_share, rep_periods)
+    end
+end
+function RepresentativePeriods(
+    len::Integer,
+    duration::S,
+    rep_periods::TimeStructure{T},
+) where {S<:Duration,T}
+    return RepresentativePeriods(
+        len,
+        duration,
+        fill(1.0 / len, len),
+        fill(rep_periods, len),
+    )
+end
+function RepresentativePeriods(
+    duration::S,
+    period_share::Vector{Float64},
```

Comment:
Could be a `Vector{<:Real}` with a `convert(Float64, period_share)`. This would allow (the unlikely) case of someone using e.g. [1, 0, 0] as a period_share.

Reply:
We have that problem in quite a few cases, among others also `op_per_strat` in `TwoLevel`... In this parameter it is however less a problem, specifically, when we consider that it should sum up to 1 and `[1, 0, 0]` does not really make sense, as it does not reduce the number of variables when iterating, just a multiplier...